File: idioms/javascript/ReadOnlyByClosure.js
#! /usr/bin/env node
"use strict";

function readonly(object, name, value){
  Object.defineProperty(object, name, {
    get: () => value
  });
}

function A() {
  readonly(this, 'constant', 'constant value');
}
const a = new A();
console.log(a.constant);

readonly(global, 'constant', 'global constant value');
console.log(constant);

constant = 5;

File: idioms/python/ReadOnlyByClosure.py
#! /usr/bin/env python

import types
import sys

def readonly(value):
    return property(lambda self: value)

class A(object):
    def __init__(self, value):
        A.constant = readonly(value)

a = A('constant value')
print(a.constant)

a.constant = 'new value'
